optimum/onnxruntime/configuration.py

```diff
@@ -965,3 +965,66 @@ def dataclass_to_dict(config) -> dict:
+        clone.quantization = self.dataclass_to_dict(clone.quantization)
+        return BaseConfig.to_dict(clone)
+
+    @classmethod
+    def from_dict(cls, config_dict: Dict[str, Any], **kwargs) -> "PretrainedConfig":
+        """
+        Overrided [`PretrainedConfig`] to make OptimizationConfig and QuantizationConfig
+
+        Args:
+            config_dict (`Dict[str, Any]`):
+                Dictionary that will be used to instantiate the configuration object. Such a dictionary can be
+                retrieved from a pretrained checkpoint by leveraging the [`~PretrainedConfig.get_config_dict`] method.
+            kwargs (`Dict[str, Any]`):
+                Additional parameters from which to initialize the configuration object.
+
+        Returns:
+            [`ORTConfig`]: The configuration object instantiated from those parameters. and `optimization` and `quantization` is each class(OptimizationConfig, QuantizationConfig).
+        """
+        return_unused_kwargs = kwargs.pop("return_unused_kwargs", False)
+        # Those arguments may be passed along for our internal telemetry.
+        # We remove them so they don't appear in `return_unused_kwargs`.
+        kwargs.pop("_from_auto", None)
+        kwargs.pop("_from_pipeline", None)
+        # The commit hash might have been updated in the `config_dict`, we don't want the kwargs to erase that update.
+        if "_commit_hash" in kwargs and "_commit_hash" in config_dict:
+            kwargs["_commit_hash"] = config_dict["_commit_hash"]
+
+        config: ORTConfig = cls(**config_dict)
+
+        if config.optimization:
+            config.optimization: OptimizationConfig = OptimizationConfig(**config.optimization)
+
+        if config.quantization:
+            quant_config: dict = config.quantization
+            quant_config["format"], quant_config["mode"] = default_quantization_parameters(
+                is_static=quant_config["is_static"],
+                format=getattr(QuantFormat, quant_config.get("format", ""), None),
+                mode=getattr(QuantizationMode, quant_config.get("mode", ""), None),
+            )
+            quant_config["activations_dtype"] = getattr(
+                QuantType, quant_config.get("activations_dtype", ""), QuantType.QUInt8
+            )
+            quant_config["weights_dtype"] = getattr(QuantType, quant_config.get("weights_dtype", ""), QuantType.QInt8)
+            config.quantization: QuantizationConfig = QuantizationConfig(**quant_config)
+
+        to_remove = []
+        for key, value in kwargs.items():
+            if hasattr(config, key):
+                setattr(config, key, value)
+                if key != "torch_dtype":
+                    to_remove.append(key)
+        for key in to_remove:
+            kwargs.pop(key, None)
+
+        logger.info(f"ORTConfig: {config.to_dict()}")
+        if return_unused_kwargs:
+            return config, kwargs
+        else:
+            return config
```

Comment:
I think a better approach would be to do:

```suggestion
    def from_dict(cls, config_dict: Dict[str, Any], **kwargs) -> "PretrainedConfig":
         config = super().from_dict(config_dict, **kwargs)
         if config.quantization:
             config.quantization = QuantizationConfig(**config.quantization)
         else:
             config.quantization = None
         # Same thing for optimization  
```

This way we duplicate less code.